pytorch_forecasting/metrics/tests/test_all_metrics.py

```diff
@@ -0,0 +1,458 @@
+        )
+
+        metric_classes.extend(
+            [
+                NormalDistributionLoss,
+                NegativeBinomialDistributionLoss,
+                LogNormalDistributionLoss,
+                MultivariateNormalDistributionLoss,
+                BetaDistributionLoss,
+                # ImplicitQuantileNetworkDistributionLoss,
+                # MQF2DistributionLoss,
+            ]
+        )
+    except ImportError:
+        raise ImportError("Failed to import distribution metrics.")
+
+    return metric_classes
+
+
+def get_test_data():
+    """Return test data fixtures for the metrics."""
+
+    batch_size, timesteps = 4, 10
+
+    test_cases = {}
+
+    y_pred_2d = torch.randn(batch_size, timesteps)
+    y_actual_2d = torch.randn(batch_size, timesteps)
+    test_cases["point"] = {"y_pred": y_pred_2d, "y_actual": y_actual_2d}
+
+    # 3d tensors for quantile forecast (default 7 quantiles)
+    y_pred_3d = torch.randn(batch_size, timesteps, 7)
+    test_cases["quantile"] = {"y_pred": y_pred_3d, "y_actual": y_actual_2d}
+
+    # data with weighted targets
+    y_actual_weighted = (y_actual_2d, torch.ones(batch_size, timesteps))
+    test_cases["weighted_2d"] = {"y_pred": y_pred_2d, "y_actual": y_actual_weighted}
+
+    test_cases["weighted_3d"] = {
+        "y_pred": y_pred_3d,
+        "y_actual": (y_actual_2d, torch.ones(batch_size, timesteps)),
+    }
+
+    # PackedSequence data (simulates variable length sequences in each batch)
+    lengths = torch.tensor([10, 8, 6, 4])
+    padded_actual = torch.randn(batch_size, timesteps)
+
+    packed_actual = rnn.pack_padded_sequence(
+        padded_actual, lengths, batch_first=True, enforce_sorted=False
+    )
+
+    test_cases["packed_sequence_2d"] = {"y_pred": y_pred_2d, "y_actual": packed_actual}
+    test_cases["packed_sequence_3d"] = {"y_pred": y_pred_3d, "y_actual": packed_actual}
+    # 3d tensors for distribution parameters
+    n_params = 4
+    y_pred_dist = torch.randn(batch_size, timesteps, n_params)
+    y_pred_dist[..., 1] = torch.abs(y_pred_dist[..., 1]) + 0.1
+    y_pred_dist[..., 3] = torch.abs(y_pred_dist[..., 3]) + 0.1
+    test_cases["distribution"] = {"y_pred": y_pred_dist, "y_actual": y_actual_2d}
+
+    n_class = 5
+    y_pred_class = torch.randn(batch_size, timesteps, n_class)
+    y_actual_class = torch.randint(0, 5, (batch_size, timesteps))
+    test_cases["classification"] = {"y_pred": y_pred_class, "y_actual": y_actual_class}
+
+    return test_cases
+
+
+DATA_FORMAT_NAMES = [
+    "point",
+    "quantile",
+    "packed_sequence_2d",
+    "packed_sequence_3d",
+    "weighted_2d",
+    "weighted_3d",
+    "distribution",
+    "classification",
+]
+
+
+@pytest.fixture(params=get_all_metrics())
+def metric_class(request):
+    """Fixture to provide metric classes."""
+    return request.param
+
+
+@pytest.fixture
+def metric_instance(metric_class):
+    if issubclass(metric_class, (MultiLoss, CompositeMetric, TorchMetricWrapper)):
```

Comment:
ok, I see we are skipping the composites for now, right?

These will probably require special handling, since the type depends on the components. In `sktime`, we use `get_test_params` and dynamic tags for cases like these.